objectivec/GPBWellKnownTypes.m

```diff
@@ -115,3 +115,52 @@ - (void)setTimeIntervalSince1970:(NSTimeInterval)timeIntervalSince1970 {
 }
 
 @end
+
+NSString *const GPBTypeGoogleApisComPrefix = @"type.googleapis.com/";
+
+@implementation GPBAny (GBPWellKnownTypes)
+
+- (NSString*)typeName {
+  NSAssert([self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix],
+           @"Invalid any type url (%@).", self.typeURL);
+  if (![self.typeURL hasPrefix:GPBTypeGoogleApisComPrefix]) return @"";
```

Comment:
If we return @"", caller doesn't need to check whether value is nil. In places that typeName are called, we can guarantee the returned value from typeName is always compared with an non-empty string.